dags/js/tasks/extract-archived-files-to-bucket.js

```diff
@@ -0,0 +1,63 @@
+const fs = require('fs');
+const Zip = require('adm-zip');
+const uuidv4 = require('uuid/v4');
+const getS3Client = require('../aws/get-s3-client');
+
+
+async function extractArchivedFilesToBucket() {
+
+  function deleteFile(fileName) {
+    fs.unlinkSync(fileName);
+    console.log('successfully deleted ' + fileName);
+  }
+
+  let tempFileName = '/tmp/' + uuidv4();
+  console.log('Temp file name =', tempFileName);
+
+  let fileStream = fs.createWriteStream(tempFileName);
+  let s3 = getS3Client();
+
+  let s3Params = {
+    Bucket: process.env.SOURCE_BUCKET,
+    Key: process.env.ARCHIVE_FILE_NAME
+  };
+
+  console.log('Getting the following from S3: ', s3Params);
+  await new Promise((resolve, reject) => {
+    s3.getObject(s3Params).createReadStream()
+      .on('end', () => {
+        return resolve();
+    }).on('error', (error) => {
+        deleteFile(tempFileName);
+        return reject(error);
+    }).pipe(fileStream)});
+
+  let archive = new Zip(tempFileName);
+  s3Params = {Bucket: process.env.DESTINATION_BUCKET};
+
+  for (let entry of archive.getEntries()) {
+    if (entry.isDirectory) {
+      continue;
+    }
+
+    // TODO: stream data to s3 asynchronously
+    s3Params.Key = process.env.ARCHIVE_FILE_NAME.replace(new RegExp('\\.\\w+$'), '') + '/' + entry.entryName;
+    s3Params.Body = entry.getData();
```

Comment:
is `getData` the synchronous part?

Reply:
yes, it returns a Buffer. I would prefer to add this to a file stream asynchronously.